File: RemoteBeesFlow/RemoteBees/JobItem.swift
/***************************************************************************
 * This source file is part of the RemoteBees open source project.         *
 *                                                                         *
 * Licensed under the MIT License. See LICENSE for license information     *
 ***************************************************************************/

public struct JobItem: Hashable, Codable, Identifiable {
    public let id: Int
    public let url: String
    public let title: String
    public let companyName: String
    public let category: String
    public let publicationDate: Date
    public let salary: String
    
    public init(
        id: Int,
        url: String,
        title: String,
        companyName: String,
        category: String,
        publicationDate: Date,
        salary: String
    ) {
        self.id = id
        self.url = url
        self.title = title
        self.companyName = companyName
        self.category = category
        self.publicationDate = publicationDate
        self.salary = salary
    }
}

File: RemoteBeesFlow/flow/src/main/java/dev/beehive/remotebees/flow/JobItem.kt
/***************************************************************************
 * This source file is part of the RemoteBees open source project.         *
 *                                                                         *
 * Licensed under the MIT License. See LICENSE for license information     *
 ***************************************************************************/

package dev.beehive.remotebees.flow

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
data class JobItem (
    val id: Int,
    val url: String,
    val title: String,
    val companyName: String,
    val category: String,
    val publicationDate: String,
    val salary : String
): Parcelable
